components/layout/query.rs

```diff
@@ -1007,17 +1014,26 @@ fn inner_text_collection_steps<N: LayoutNode>(node: N,
             },
             _ => {},
         }
+
+        results.append(&mut items);
+        inner_text_collection_steps(iter, indexable_text, results);
     }
+}
 
-    results.append(&mut items);
+fn is_last_table_cell<N: LayoutNode>(node: &N) -> bool {
+    !has_next_sibling(node, Display::TableCell)
 }
 
-fn is_last_table_cell() -> bool {
-    // FIXME(ferjm) Implement this.
-    false
+fn is_last_table_row<N: LayoutNode>(node: &N) -> bool {
+    !has_next_sibling(node, Display::TableRow)
 }
 
-fn is_last_table_row() -> bool {
-    // FIXME(ferjm) Implement this.
-    false
+fn has_next_sibling<N: LayoutNode>(node: &N, display: Display) -> bool {
+    node.parent_node().map_or(false, |parent| {
```

Comment:
So, this is very confusing. Why is the function called `has_next_sibling`, yet the function looks at the `parent`'s next sibling?

Also, I'm pretty sure treating text-nodes as having the display value of the parent is wrong, since they're always inline effectively.